src/wallet/db.cpp

```diff
@@ -614,7 +614,9 @@ void BerkeleyBatch::Flush()
     if (fReadOnly)
         nMinutes = 1;
 
-    env->dbenv->txn_checkpoint(nMinutes ? gArgs.GetArg("-dblogsize", DEFAULT_WALLET_DBLOGSIZE) * 1024 : 0, nMinutes, 0);
+    if (env) {
```

Comment:
Why this change?

Reply:
So that tests pass. When there's a dummy database, env will be `nulltpr` and this causes a segfault in tests which use a dummy database.

Reply:
Can't we make `env` is defined instead? Otherwise a comment would make sense as this doesn't really happen-

Reply:
> Can't we make `env` is defined instead?

No. `env` is nullptr for dummy databases which are used in tests. A check like this is used in many other places in this file in the form of `BerkeleyDatabase::IsDummy()`.

> Otherwise a comment would make sense as this doesn't really happen.

Done.

Reply:
nit: could deal with it earlier, and use `IsDummy` to make it more explicit.

Reply:
`IsDummy` is done by `BerkeleyDatabase`, not `BerkeleyBatch` so is not available to do here (although the check is completely identical).